packages/popup/src/ux-popup.ts

```diff
@@ -0,0 +1,170 @@
+import { customElement, inject, bindable, PLATFORM, TaskQueue, useView, DOM } from 'aurelia-framework';
+import { UxComponent, StyleEngine, normalizeBooleanAttribute } from '@aurelia-ux/core';
+import { UxPopupTheme } from './ux-popup-theme';
+
+const windowEvents = ['click', 'wheel', 'scroll', 'resize'];
+
+interface IAnchor {
+  left: string | undefined;
+  right: string | undefined;
+  top: string | undefined;
+  bottom: string | undefined;
+  maxHeight: number | undefined;
+  maxWidth: number | undefined;
+}
+
+@inject(Element, StyleEngine, TaskQueue)
+@customElement('ux-popup')
+@useView(PLATFORM.moduleName('./ux-popup.html'))
+export class UxPopup implements UxComponent, EventListenerObject {
+  constructor(private element: HTMLElement, private styleEngine: StyleEngine, private taskQueue: TaskQueue) { }
+
+  public anchor: IAnchor | null;
+  public isWrapperOpen: boolean = false;
+  public isOpen: boolean = false;
+  public popupContent: HTMLDivElement;
+  public isMeasured: boolean = false;
+
+  @bindable
+  trigger: HTMLElement | string | null;
+  triggerChanged(newValue: HTMLElement | string, oldValue: HTMLElement | string) {
+    if (typeof (newValue) === 'string') {
+      this.taskQueue.queueTask(() => this.trigger = DOM.querySelector(newValue) as HTMLElement);
```

Comment:
I think this will make the functionality a bit brittle, any trigger must be added before this popup, otherwise it's gonna fail. I think we will need to keep it as is if it's a string, and perform a single check inside `onWindowClick`